### PR TITLE
local_maillog feature: Notification of queuemails setting.

### PR DESCRIPTION
**Pull request by sumaiyamannan**
A page is created to show the queuemails setting status.

### CODE REVIEW

File: lib.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+/**
+ * Mail log plugin by Catalyst IT.
+ * Update Moodle user bounce counts and delete mail message
+ * Delete old bounce messages
+ * Expunge
+ *
+ * @package local_maillog
+ * @author  Sumaiya Javed <sumaiya.javed@catalyst.net.nz>
+ * @license http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
```
Comment (danmarsden) [inline lib.php:26]: can you please drop in the copyright catalyst line here?

File: lib.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+/**
+ * Records the date of the last update to queuemails admin setting.
+ *
+ * @return void
+ */
+function local_maillog_notify() {
+    // Add time when the queuemails setting was updated.
+    global $DB;
+    $name = 'queuemailsdate'; $plugin = 'local_maillog'; $now = time();
```
Comment (danmarsden) [inline lib.php]: bit weird seeing these all on one line - unless I've missed a new coding guideline somewhere lets drop them onto their own lines.

File: lib.php [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    $id = $DB->get_field('config_plugins', 'id', array('name' => $name, 'plugin' => $plugin));
```
Comment (danmarsden) [inline lib.php]: is there a reason this isn't using the get_config() api ? - could end up with weird behaviour as it's also cached.

File: notify.php [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+header('Cache-Control: private, no-cache, no-store, max-age=0, must-revalidate, proxy-revalidate');
+header('Expires: Tue, 04 Sep 2012 05:32:29 GMT');
+global $DB;
+$format = '%b %d %H:%M:%S';
+$now = userdate(time(), $format);
+
+$queuemails = get_config('local_maillog', 'queuemails');
+if ($queuemails) {
+    $name = 'queuemailsdate'; $plugin = 'local_maillog';
+    $value = $DB->get_field('config_plugins', 'value', array('name' => $name, 'plugin' => $plugin));
```
Comment (danmarsden) [inline notify.php]: should be using get_config() api here.

File: notify.php [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Mail log plugin by Catalyst IT.
+ * Notifies if the setting of queuemails is switched on.
+ *
+ * @package local_maillog
+ * @author  Sumaiya Javed <sumaiya.javed@catalyst.net.nz>
```
Comment (danmarsden) [inline notify.php:22]: missing copyright phpdoc line here too.

File: notify.php [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+define('NO_UPGRADE_CHECK', true);
+$dirroot = __DIR__ . '/../../';
+define('NO_MOODLE_COOKIES', true);
+
+require($dirroot.'config.php');
+header("Content-Type: text/plain");
+header('Pragma: no-cache');
+header('Cache-Control: private, no-cache, no-store, max-age=0, must-revalidate, proxy-revalidate');
+header('Expires: Tue, 04 Sep 2012 05:32:29 GMT');
+global $DB;
```
Comment (danmarsden) [inline notify.php]: global $DB shouldn't be needed here - it's already availiable directly after including config.php

File: notify.php [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+ * @package local_maillog
+ * @author  Sumaiya Javed <sumaiya.javed@catalyst.net.nz>
+ * @license http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+define('NO_UPGRADE_CHECK', true);
+$dirroot = __DIR__ . '/../../';
+define('NO_MOODLE_COOKIES', true);
+
+require($dirroot.'config.php');
```
Comment (danmarsden) [inline notify.php]: can we just use:
```
require(__DIR__ . '/../../config.php');
```
(bit cleaner and more typical of other files.)

File: lib.php (hunk 3 above)
Comment (sumaiyamannan) [reply lib.php]: I see a single set_config() is doing the job. Thanks for your feedback.

File: notify.php [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+$format = '%b %d %H:%M:%S';
+$now = userdate(time(), $format);
+$queuemails = get_config('local_maillog', 'queuemails');
+if ($queuemails) {
+    $queuemailsdate = get_config('local_maillog', 'queuemailsdate');
+    $hourspassed = round((time() - $queuemailsdate) / 3600);
+    $msg = get_string('pluginname', 'local_maillog')." ".get_string('queuemails', 'local_maillog').": switched on. Hours passed: ".$hourspassed;
+    printf ("CRITICAL: $msg (Checked $now)\n");
+}
+
```
Comment (matt-catalyst) [inline notify.php:46]: Can it return 'OK' if the queuemails is not set.

**Review by matt-catalyst — APPROVED** [review]

**Pull request merged**